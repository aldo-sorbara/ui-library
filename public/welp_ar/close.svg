<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
