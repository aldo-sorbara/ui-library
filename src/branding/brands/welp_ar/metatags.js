export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
