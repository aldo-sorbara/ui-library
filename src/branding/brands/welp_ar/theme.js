import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
