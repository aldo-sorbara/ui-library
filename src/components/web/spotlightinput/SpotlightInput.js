/* stylelint-disable order/properties-alphabetical-order */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import Input from '@bit/wenance.common-ui.web.input';
import { Spring } from 'react-spring/renderprops.cjs.js';
import styled from 'styled-components';

export const SpotlightInput = props => {
  const { name, data, info } = props;
  const [focus, setFocus] = useState(false);
  const handleKeypress = e => {
    if (e.keyCode === 13 || e.keyCode === 27) {
      focus && setFocus(false);
    }
  };
  useEffect(() => {
    document.addEventListener('keydown', handleKeypress, true);
    return () => {
      document.removeEventListener('keydown', handleKeypress, true);
    };
  });
  return (
    <Spring
      from={{
        fontSize: '1rem',
        borderColor: 'white',
        margin: '0.5rem 0',
        padding: 0
      }}
      to={{
        fontSize: focus ? '3rem' : '1rem',
        borderColor: focus ? 'black' : 'white',
        padding: focus ? 16 : 0
      }}
    >
      {({ fontSize, t, ...rest }) => (
        <Container style={{ ...rest }}>
          <Input
            {...props}
            style={{ fontSize }}
            data={data || name}
            onFocus={e => {
              setFocus(true);
            }}
            size={focus ? 'xLarge' : 'small'}
            onBlur={e => {
              setFocus(false);
            }}
          />
          {focus && info}
        </Container>
      )}
    </Spring>
  );
};

const Container = styled.div`
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
`;

SpotlightInput.defaultProps = {
  theme: {
    typography: {
      label: {
        default: {
          fontSize: '1rem',
          fontWeight: '700',
          lineHeight: '1.33'
        }
      }
    },
    palette: {
      secondary: {
        light: 'inherit'
      },
      common: {
        text: {
          default: '#f3a'
        }
      }
    }
  }
};

SpotlightInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  data: PropTypes.string,
  info: PropTypes.string
};

export default SpotlightInput;
