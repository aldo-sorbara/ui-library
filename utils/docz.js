export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
