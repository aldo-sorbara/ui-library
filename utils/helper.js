import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
