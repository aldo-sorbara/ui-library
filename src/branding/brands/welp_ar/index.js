import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
