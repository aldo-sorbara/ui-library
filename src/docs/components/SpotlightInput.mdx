---
name: SpotlightInput
menu: Components
route: /spotlightinput
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import SpotlightInput from '../../components/web/spotlightinput';
import Input from '../../components/web/input';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# SpotlightInput

<PropsTable of={SpotlightInput} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Input
        name='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        descriptionBottom
        description='celular'
      />
      <SpotlightInput
        name='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        descriptionBottom
        autoComplete={'off'}
        description='celular'
        info='Te enviaremos un codigo por SMS'
      />
      <Input
        name='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        descriptionBottom
        description='celular'
      />
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/input--with-prefix-label-and-placeholder)
