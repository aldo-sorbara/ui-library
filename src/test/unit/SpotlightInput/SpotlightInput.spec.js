import React from 'react';
import SpotlightInput from '../../../components/web/spotlightinput';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import { act } from 'react-dom/test-utils';

describe('Input component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <SpotlightInput
          name='phone'
          data='phone'
          label='Ingrese su celular'
          placeholder='42424242'
          prefix='15'
          info='para enviarte un sms'
          meta={{ active: true, valid: true }}
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('focus and blur input', () => {
    const wrapper = mountWithTheme(
      <SpotlightInput
        name='phone'
        data='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        info='para enviarte un sms'
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    wrapper.find('input').simulate('focus');
    wrapper.find('input').simulate('blur');
  });
  it('should close modal on press enter', () => {
    const wrapper = mountWithTheme(
      <SpotlightInput
        name='phone'
        data='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        info='para enviarte un sms'
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    expect(wrapper.find('div').length).toBe(4);
    wrapper.find('input').simulate('focus');
    expect(wrapper.find('div').length).toBe(4);
    let eventEnter = new KeyboardEvent('keydown', { keyCode: 13 });
    act(() => {
      document.dispatchEvent(eventEnter);
    });
    setTimeout(() => {
      expect(wrapper.find('div').length).toBe(4);
    }, 1500);
  });
  it('should close modal on press esc', () => {
    const wrapper = mountWithTheme(
      <SpotlightInput
        name='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        info='para enviarte un sms'
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    expect(wrapper.find('div').length).toBe(4);
    wrapper.find('input').simulate('focus');
    expect(wrapper.find('div').length).toBe(4);
    let eventEsc = new KeyboardEvent('keydown', { keyCode: 27 });
    act(() => {
      document.dispatchEvent(eventEsc);
    });
    setTimeout(() => {
      expect(wrapper.find('div').length).toBe(4);
    }, 1500);
  });
  it('should do nothing on keycode 30', () => {
    const wrapper = mountWithTheme(
      <SpotlightInput
        name='phone'
        label='Ingrese su celular'
        placeholder='42424242'
        prefix='15'
        info='para enviarte un sms'
        meta={{ active: true }}
      />,
      welp_ar.theme
    );
    wrapper.find('input').simulate('focus');
    let eventEsc = new KeyboardEvent('keydown', { keyCode: 30 });
    act(() => {
      document.dispatchEvent(eventEsc);
    });
  });
});
