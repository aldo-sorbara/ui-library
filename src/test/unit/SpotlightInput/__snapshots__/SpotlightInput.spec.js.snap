// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render snapshot 1`] = `
.c1 {
  width: 100%;
}

.c4 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  font-weight: normal;
  padding-right: .25rem;
  text-align: center;
  white-space: nowrap;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  border: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  border: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  border: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  border: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;
}

.c5:focus {
  outline: none;
}

.c5:-webkit-autofill,
.c5:-webkit-autofill:hover,
.c5:-webkit-autofill:focus {
  background-color: transparent !important;
  -webkit-box-shadow: none !important;
  -webkit-text-fill-color: #000;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
}

.c2 {
  display: block;
  font-weight: 700;
  line-height: 1;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  padding-bottom: .25rem;
}

.c3 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 1rem;
  font-weight: 300;
  margin-bottom: .25rem;
  padding: .25rem .25rem .25rem 0;
  line-height: 1.5;
  position: relative;
  width: 100%;
}

.c7 {
  bottom: 0;
  display: block;
  height: 2px;
  left: 0;
  position: absolute;
  right: 0;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  white-space: nowrap;
}

.c6 > svg {
  display: block;
  height: 1.125rem;
  width: 1.125rem;
}

.c0 {
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
}

<div
  className="c0"
  style={
    Object {
      "borderColor": "rgba(255, 255, 255, 1)",
      "margin": "0.5rem 0",
      "padding": 0,
    }
  }
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Ingrese su celular
       
    </label>
    <div
      className="c3"
      size="small"
    >
      <span
        className="c4"
        size="small"
      >
        15
      </span>
      <input
        className="c5"
        data="phone"
        data-w="phone"
        label="Ingrese su celular"
        name="phone"
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="42424242"
        prefix="15"
        size="small"
        style={
          Object {
            "fontSize": "1rem",
          }
        }
      />
      <span
        className="c6"
        size="small"
      />
      <div
        className="c7"
        data="phone"
        label="Ingrese su celular"
        name="phone"
        onBlur={[Function]}
        onChange={null}
        onFocus={[Function]}
        placeholder="42424242"
        prefix="15"
        size="small"
        style={
          Object {
            "fontSize": "1rem",
          }
        }
      />
    </div>
  </div>
</div>
`;
